### PR TITLE
Fixup shared spaces

### PR DESCRIPTION
**Pull request by JBludau**
Fixup for: 

- discussion https://github.com/kokkos/kokkos/pull/5405#discussion_r982473816 (continued in the dev-meeting + decision to only define the symbols `KOKKOS_HAS_SHARED_SPACE` and `KOKKOS_HAS_SHARED_HOST_PINNED_SPACE` but not define them to `1` for enabled. Decision mainly to keep it consistent in Kokkos). 
- Removed some unnecessary brackets in cmake. 
- Fixed bug in `TestSharedSpace.cpp`, it now compares to the right timings on the host.

### CODE REVIEW

File: core/unit_test/TestSharedSpace.cpp [hunk 1]
```diff
@@ -156,62 +155,57 @@ TEST(defaultdevicetype, shared_space) {
...
-      incrementInLoop<Kokkos::DefaultHostExecutionSpace>(defaultHostExecData,
-                                                         numRepetitions);
+  auto hostLocalTimings =
+      incrementInLoop<HostExecutionSpace>(hostData, numRepetitions);
 
   // GET PAGE MIGRATING TIMINGS DATA
-  std::vector<decltype(defaultExecLocalTimings)> defaultExecSharedTimings{};
-  std::vector<decltype(defaultExecLocalTimings)> defaultHostExecSharedTimings{};
+  std::vector<decltype(deviceLocalTimings)> deviceSharedTimings{};
+  std::vector<decltype(deviceLocalTimings)> hostSharedTimings{};
```
Comment (dalg24) [inline core/unit_test/TestSharedSpace.cpp:180]: I know they are the same type but I'd prefer 
```suggestion
  std::vector<decltype(hostLocalTimings)> hostSharedTimings{};
```

File: core/unit_test/TestSharedSpace.cpp [hunk 2]
```diff
@@ -156,62 +155,57 @@ TEST(defaultdevicetype, shared_space) {
...
+                  deviceSharedTimings[cycle].end(), [&](const uint64_t timing) {
+                    (timing < threshold * deviceLocalMean)
                         ? fastAsLocalOnRepeatedAccess &= true
                         : fastAsLocalOnRepeatedAccess &= false;
                   });
 
-    std::for_each(std::next(defaultHostExecSharedTimings[cycle].begin()),
-                  defaultHostExecSharedTimings[cycle].end(),
-                  [&](const uint64_t timing) {
-                    (timing < threshold * defaultExecLocalMean)
```
Comment (dalg24) [inline core/unit_test/TestSharedSpace.cpp:214]: So that was the issue?  `defaultExecLocalMean` instead of `hostLocalMean` in the line above?

Comment (JBludau) [reply core/unit_test/TestSharedSpace.cpp:214]: yup ... it was comparing to device timings instead of host timings but host timings is what we wanted

**Review by dalg24 — APPROVED** [review]

**Comment by janciesko** [comment]
LGTM

**Review by janciesko — APPROVED** [review]

**Pull request merged**